src/azure-cli/azure/cli/command_modules/botservice/tests/latest/test_bot_commands.py

```diff
@@ -809,6 +809,97 @@ def test_botservice_create_should_remove_invalid_char_from_name_when_webapp(self
                      self.check('type', 'abs')
                  })
 
+    @ResourceGroupPreparer(random_name_length=20)
+    def test_botservice_create_enable_encryption_then_disable(self, resource_group):
+        self.kwargs.update({
+            'botname': self.create_random_name(prefix='cli', length=15),
+            'app_id': str(uuid.uuid4()),
+            'display_name': 'clitestbot',
+            'keyvaultname': self.create_random_name(prefix='keyVault', length=15),
+            'keyName': self.create_random_name(prefix='keyName', length=15),
+        })
+
+        self.cmd('az bot delete -g {rg} -n {botname}')
+
+        self.cmd('az keyvault create --location westus2 --name {keyvaultname} -g {rg}')
+        self.cmd(
+            'az keyvault update --name {keyvaultname} -g {rg} --enable-soft-delete true --enable-purge-protection true')
+        self.cmd(
+            'az keyvault set-policy --name {keyvaultname} --key-permissions get wrapKey unwrapKey --object-id a5c54cac-9ebb-45c5-81e5-29dabf2eb5ee')
+
+        resultAzKey = self.cmd(
+            'az keyvault key create --vault-name {keyvaultname} --name {keyName} --protection software',
+            checks={
+                self.check('key', object)
+            })
+        resultAzKey = resultAzKey.get_output_in_json()
+        cmk_url = resultAzKey['key']['kid']
```

Comment:
@RustyDes or @swagatmishra2007, does Azure Bot Service do any validation on this KeyVault URL and key when they're provided?

If yes, does ABS look for uniqueness in the URL and/or key? Or can multiple bots use the same values. 

If uniqueness isn't a requirement, the KeyVault values could be provisioned once and reused for each test as a pre-arrangement function.

Reply:
There's validation done, but it can be reuse by multiple bots (in fact I could easily imagine a company with more than one bot having a single key that they use for all their bots and possibly some other part of their systems such as their own storage - remember other azure features also support cmk, and it would make sense for the customer to use 1 key for all of them).

Reply:
Thanks @RustyDes! So @munozemilio it could speed up the tests to arrange a single key vault instance with a key than provisioning a new key vault resource for every test.

@yonzhan what do you think on this proposed test refactoring?

Reply:
@fengzhou-msft could you help with this?

Reply:
@stevengum I did a refactor to create the Key Vault resource and key during the setup but started to get this error: Key not found. Ensure that the key exist (if it was deleted it may need to be undeleted).
This is not thrown if created within the same test. Given the deadline is almost here I will refactor that once I find a most efficient and successful way to achieve that.

Reply:
Understood, thanks for giving it a go @munozemilio!